_posts/2024-14-18-asp-react-docker-compose.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: distill
+title: Asp dot net 8.0 and React with Docker-compose
+description: How to create a project with Asp Dotnet 8.0 and React with Docker-compose
+tags: Docker-compose With dotnet and react
+giscus_comments: true
+date: 2024-04-18
+featured: false
+
+authors:
+  - name: Usman Jamil Bhatti
+    url: "https://github.com/usman-jamil"
+    affiliations:
+      name: Allied Consultants
+  - name:  Sheikh Luqman
+    url: "https://en.wikipedia.org/wiki/Boris_Podolsky"
+    affiliations:
+      name: Allied Consultants
+  - name: Zuhaib Sohail
+    url: "https://github.com/ZuhaibSohail163"
+    affiliations:
+      name: Allied Consultants
+
+_styles: >
+  .fake-img {
+    background: #bbb;
+    border: 1px solid rgba(0, 0, 0, 0.1);
+    box-shadow: 0 0px 4px rgba(0, 0, 0, 0.1);
+    margin-bottom: 12px;
+  }
+  .fake-img p {
+    font-family: monospace;
+    color: white;
+    text-align: left;
+    margin: 12px 0;
+    text-align: center;
+    font-size: 16px;
+  }
+
+---
+
+ASP Dot Net 8.0 and React.ts with Docker compose
+Create root folder name like `asp-react-nginx-compose` or other suit able.
+
+In it create 3 folders and 1 file in root folder :
+#### The Project structure is:
+```plaintext
+app
+├── backend
+├── docker-compose.yml
+├── frontend
+└── nginx
+
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/Luqmant51/asp-react-nginx-compose) of given example project asp-react-nginx-compose.
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
+Before you start setting up your project, you need to install Docker on your machine if it isn't already installed. Docker allows you to create, deploy, and run applications by using containers.
```

Comment:
Add: We will see how `docker compose` can be used to perform orchestration between containers, keeping isolation in mind i.e., while some containers can talk to each other, we might not want that for others e.g., have NGINX be the communications broker between the front-end and back-end and not allow them to talk to each other directly. This has several use-cases which are outside the scope of this blog.

Reply:
done